vault/request_handling.go

```diff
@@ -126,7 +126,7 @@ func (c *Core) fetchACLTokenEntryAndEntity(ctx context.Context, req *logical.Req
 
 	// Ensure there is a client token
 	if req.ClientToken == "" {
-		return nil, nil, nil, nil, &logical.StatusBadRequest{Err: "missing client token"}
+		return nil, nil, nil, nil, logical.ErrPermissionDenied
```

Comment:
I'm still of the opinion that we don't want to return the generic `PermissionDenied` error is actually less descriptive than a `missing client token` message, which more accurately captures what is going on here. I'm happy to let this go in if others think differently, but I'm of the opinion that we should more towards more descriptive error messages in each clause as opposed to consistently throwing a message like `logical.ErrPermissionDenied` no matter which error clause of a function the error occurs in.

Reply:
Hmm I think I'm also in favor of keeping the more descriptive `missing client token` error here. Also, I don't think this really addresses the issue reported, imo we would either want to change https://github.com/hashicorp/vault-enterprise/blob/main/vault/router.go#L810 to return true for a login attempt on a non-existent mount, or close the issue out as a won't do.

Reply:
( retracting my comment around the PR not addressing the issue, after going through slack conversation )